stdlib/private/StdlibUnittest/StdlibUnittest.swift

```diff
@@ -1447,7 +1450,12 @@ public func runAllTests() {
   if _isChildProcess {
     _childProcess()
   } else {
+    #if os(WASI)
+    // WASI doesn't support child processes
+    var runTestsInProcess: Bool = true
```

Comment:
This won't work.  The death tests need to be run as child processes.  I think it is better to not make the test suite harness more complicated for something which won't work.  That is drop the changes to `stdlib/private/StdlibUnittest`